internal/provider/inventory_data_source_test.go

```diff
@@ -1,269 +1,204 @@
-						},
-					},
-				},
-			},
-			expected: inventoryDataSourceModel{
-				ID:     basetypes.NewInt64Value(1),
-				Groups: map[string]groupDataSourceModel{},
-				Hosts: map[string]hostDataSourceModel{
-					"test": {
-						HostVars: map[string]string{
-							"some_var": "some_var_value",
-						},
-					},
-				},
+			name: "all values",
+			input: []byte(
+				`{"id":1,"organization":2,"url":"/inventories/1/","name":"my inventory","description":"My Test Inventory","variables":"{\"foo\":\"bar\"}"}`,
+			),
+			expected: InventoryDataSourceModel{
+				Id:           types.Int64Value(1),
+				Organization: types.Int64Value(2),
+				Url:          types.StringValue("/inventories/1/"),
+				Name:         types.StringValue("my inventory"),
+				Description:  types.StringValue("My Test Inventory"),
+				Variables:    jsontypes.NewNormalizedValue("{\"foo\":\"bar\"}"),
 			},
+			errors: diag.Diagnostics{},
 		},
 	}
-	for _, tc := range testTable {
-		t.Run(tc.name, func(t *testing.T) {
-			tc.state.addHostVariable("test", "some_var", "some_var_value")
-			if !reflect.DeepEqual(tc.state, tc.expected) {
-				t.Errorf("expected (%v)", tc.expected)
-				t.Errorf("result   (%v)", tc.state)
+
+	for _, test := range testTable {
+		t.Run(test.name, func(t *testing.T) {
+			resource := InventoryDataSourceModel{}
+			diags := resource.ParseHttpResponse(test.input)
+			if !test.errors.Equal(diags) {
+				t.Errorf("Expected error diagnostics (%s), Received (%s)", test.errors, diags)
+			}
+			if test.expected != resource {
+				t.Errorf("Expected (%s) not equal to actual (%s)", test.expected, resource)
 			}
 		})
 	}
 }
 
-func TestMapHosts(t *testing.T) {
-	testTable := []struct {
-		name     string
-		hosts    []ansibleHost
-		state    inventoryDataSourceModel
-		expected inventoryDataSourceModel
-	}{
-		{
-			name: "case 1",
-			hosts: []ansibleHost{
-				{
-					Name:   "web",
-					Groups: []string{"deployer", "front"},
-					Variables: map[string]string{
-						"framework": "django",
-					},
-				},
-				{
-					Name:   "db",
-					Groups: []string{"deployer", "database"},
-					Variables: map[string]string{
-						"server":  "postgresql",
-						"version": "14.0.0",
-					},
-				},
-				{
-					Name:      "ansible",
-					Groups:    []string{},
-					Variables: map[string]string{},
-				},
-			},
-			state: inventoryDataSourceModel{
-				ID:     basetypes.NewInt64Value(1),
-				Groups: nil,
-				Hosts:  nil,
+func TestAccInventoryDataSource(t *testing.T) {
+	var inventory InventoryAPIModel
+	randomName := acctest.RandStringFromCharSet(10, acctest.CharSetAlphaNum)
+	updatedName := "Inventory " + randomName
+	updatedDescription := "Test inventory"
+	updatedVariables := "{\"foo\": \"bar\"}"
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { testAccPreCheck(t) },
+		ProtoV6ProviderFactories: testAccProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			// Create and Read testing
```

Comment:
For the data source acceptance tests we don't need to test create or update since data sources don't create anything in AAP. Instead, I would suggest creating an inventory in step 1 (using the inventory resource config) and then doing a read test in step 2 with a data source config, using the id from the inventory created in step 1. In step 2 we don't need to check that the resource exists, just that the data source attribute values match what we expect. Let me know if you have questions about any of that.

Reply:
From my understanding, I have addressed your comments. Please let me know if this aligns with your expectations.